### PR TITLE
Added upload progressbar to the LTI upload tool

### PR DESCRIPTION
**Pull request by kristofkeppens**
Adding a simple uploadprogress to the upload form when uploading via LTI

### CODE REVIEW

File: modules/lti/package.json [hunk 1]
```diff
@@ -9,10 +9,14 @@
     "@fortawesome/react-fontawesome": "^0.1.14",
     "axios": "^0.21.1",
     "bootstrap": "^4.6.0",
+    "dangerously-set-html-content": "^1.0.8",
+    "engage-ui": "../engage-ui",
+    "http-mock-server": "^0.5.2",
```
Comment (lkiesow) [inline modules/lti/package.json]: This is already listed as devDependency. It's probably not necessary to listr this as runtime dependency as well.

File: modules/lti/src/components/Upload.tsx [hunk 2]
```diff
@@ -108,6 +111,42 @@ class TranslatedUpload extends React.Component<UploadProps, UploadState> {
...
+                    },
+                });
+            }
+        }).catch((_) => {
+            console.log("couldn't retrieve metadata at this time, ignoring...");
+        });
+    }
+
+    setUploadPogress(progress: number) {
+        console.log(progress)
```
Comment (lkiesow) [inline modules/lti/src/components/Upload.tsx]:
```suggestion
        console.debug(progress)
```

File: modules/lti/src/components/Upload.tsx [hunk 3]
```diff
@@ -108,6 +111,42 @@ class TranslatedUpload extends React.Component<UploadProps, UploadState> {
         }));
     }
 
+    componentWillUnmount() {
```
Comment (lkiesow) [inline modules/lti/src/components/Upload.tsx]: You removed this block in pull request #2263
This probably shouldn't be in here, should it?

File: modules/lti/package.json [hunk 4]
```diff
@@ -9,10 +9,14 @@
     "@fortawesome/react-fontawesome": "^0.1.14",
     "axios": "^0.21.1",
     "bootstrap": "^4.6.0",
+    "dangerously-set-html-content": "^1.0.8",
+    "engage-ui": "../engage-ui",
```
Comment (lkiesow) [inline modules/lti/package.json]: What do we need to include the `engage-ui` module for?

**Review by lkiesow — CHANGES_REQUESTED** [review]
Initial review without a lot of testing.

File: modules/lti/package.json (hunk 1 above)
Comment (kristofkeppens) [reply modules/lti/package.json]: These dependencies came from the deeplink PR and are not needed here, removed them.

**Comment by Arnei** [comment]
This might just be me failing to resolve the conflicts properly, but I can't build the lti module with this PR.

```
[INFO] Failed to compile.
[INFO] 
[INFO] ./src/components/Series.tsx
[INFO] Cannot find module: '/home/arnewilken/IdeaProjects/opencast9/modules/lti/node_modules/babel-preset-react-app/node_modules/@babel/runtime/helpers/esm/objectSpread2'. Make sure this package is installed.
```

**Comment by kristofkeppens** [comment]
@Arnei I fixed the merge conflicts and you should be able to build it now.

**Review by Arnei — CHANGES_REQUESTED** [review]
Works fine in my local OC AllInOne installation and code looks fine too.

Regarding the UI, I would prefer it if the percentage number would not move with the progress bar. It does not add any additional information and can make it stressful to read. Although I'm not a professional UI designer, so if people say it is fine the way it is I'm fine with conceding this request.

It would also be nice if the "Upload successful" message was not in a completely different place from the upload progress bar, but that is arguably not this PRs fault and not a change I am requesting for this PR.

**Comment by kristofkeppens** [comment]
@Arnei we are using the react-bootstrap progressbar ( https://react-bootstrap.github.io/components/progress/ ), without heavy modification of this it's not able to change the way the percentage is shown. We could remove it and only keep it for screen-readers, if this is preferred ?

**Comment by KatrinIhler** [comment]
I talked to Arne and he was fine with the current state if it can't be easily changed. So I'm gonna go ahead and merge this. :)

**Pull request merged**